src/DeviceInterface.h

```diff
@@ -9,18 +9,17 @@
 
 namespace Halide {
 
-/** Get the appropriate halide_device_interface_t * for a
- * target. Corresponds to the device interface that would be used for
- * DeviceAPI::Default_GPU. Creates a GPU runtime module for the target
- * if necessary. Returns nullptr if no device APIs are enabled in the
- * target. */
-EXPORT const halide_device_interface_t *get_default_device_interface_for_target(const Target &t);
-
 /** Gets the appropriate halide_device_interface_t * for a
- * DeviceAPI. Returns null if that device API is not enabled in the
- * target, or if the argument is None or Host. */
-EXPORT const halide_device_interface_t *get_device_interface_for_device_api(const DeviceAPI &d,
-                                                                            const Target &t = get_jit_target_from_environment());
+ * DeviceAPI. If error_site is non-null, e.g. the name of the routine
+ * calling get_device_interface_for_device_api, a user_error is
+ * reported if the requested device API is not enabled in or supported
+ * by the target, Halide has been compiled without this device API, or
+ * the device API is None or Host or a bad value. The error_site
+ * argument is printed in the error message. If error_site is null,
+ * this routine returns nullptr instead of calling user_error. */
+EXPORT const halide_device_interface_t *get_device_interface_for_device_api(DeviceAPI d,
+                                                                            const Target &t = get_jit_target_from_environment(),
+                                                                            const char *error_site = nullptr);
```

Comment:
I think this should have a non-null default, and maybe the function itself should assert it's non-null so we don't give silent failures.

Reply:
The return null behavior is used in at least one test to skip testing the particular GPU API if it is not enabled.